ToolWrapperLayer/HISAT2Wrapper.cs

```diff
@@ -43,5 +44,60 @@ public string WriteRemoveScript(string spritzDirectory)
         }
 
         #endregion Installation Methods
+        #region Alignment Methods
+
+        public static bool IndexExists(string genomeFasta)
+        {
+            return File.Exists(Path.Combine(Path.GetDirectoryName(genomeFasta), Path.GetFileNameWithoutExtension(genomeFasta) + ".1.ht2"));
+        }
+
+        public static void GenerateIndex(string spritzDirectory, string analysisDirectory, string genomeFasta, out string IndexPrefix)
+        {
+            IndexPrefix = Path.Combine(Path.GetDirectoryName(genomeFasta), Path.GetFileNameWithoutExtension(genomeFasta));
+            if (IndexExists(genomeFasta))
+            {
+                return;
+            }
+            WrapperUtility.GenerateAndRunScript(WrapperUtility.GetAnalysisScriptPath(analysisDirectory,"Hisat2Build.bash"), new List<string>
+            {
+                WrapperUtility.ChangeToToolsDirectoryCommand(spritzDirectory),
+                "hisat2-2.1.0/hisat2-build" + 
+                    " " + WrapperUtility.ConvertWindowsPath(genomeFasta) +
+                    " " + WrapperUtility.ConvertWindowsPath(IndexPrefix)
+            }).WaitForExit();
+        }
+
+        public static void Align(string spritzDirectory, string analysisDirectory, string IndexPrefix, string[] fastqPaths, out string outputDirectory)
+        {
+            if (fastqPaths.Count() == 1)
```

Comment:
Use `.Length` instead (and `.Count` for list objects). `Count()` is a Linq function that is slow, especially in loops.